worlds/tunic/__init__.py

```diff
@@ -412,12 +429,34 @@ def fill_slot_data(self) -> Dict[str, Any]:
             "disable_local_spoiler": int(self.settings.disable_local_spoiler or self.multiworld.is_race),
         }
 
+        # this would be in a stage if there was an appropriate stage for it
+        if not self.item_link_locations:
+            tunic_worlds: Tuple[TunicWorld] = self.multiworld.get_game_worlds("TUNIC")
+            # figure out our groups and the items in them
+            for tunic in tunic_worlds:
+                for group in self.multiworld.get_player_groups(tunic.player):
+                    self.item_link_locations.setdefault(group, {})
+            for location in self.multiworld.get_locations():
+                if location.item.player in self.item_link_locations.keys():
+                    self.item_link_locations[location.item.player].setdefault(location.item.name, [])
+                    self.item_link_locations[location.item.player][location.item.name].append((location.player, location.name))
```

Comment:
A problem with the `if not self.item_link_locations` check is that if no TUNIC players belong to a single group (so there are no TUNIC players with item links), then `self.item_link_locations` will always be empty, so every TUNIC world will enter this conditional branch. This can be resolved by having `item_link_locations` start off as `None` and changing the check to `if self.item_link_locations is None:`.

Also, in the case of no TUNIC players belonging to a group, it is unnecessary to check the locations in the multiworld because `self.item_link_locations` will be empty, meaning that `location.item.player in self.item_link_locations.keys()` will always be `False`.

And `dict.setdefault()` returns the value (either the existing value if there was one, or the new value if there was not), so the last two lines can be simplified a bit by storing the result of `self.item_link_locations[location.item.player].setdefault(location.item.name, [])` into a variable.

```suggestion
        if self.item_link_locations is None:
            item_link_locations = {}
            # item_link_locations is a class attribute shared by all TunicWorld instances, so must be set on the class
            # and not per-instance.
            TunicWorld.item_link_locations = item_link_locations
            tunic_worlds: Tuple[TunicWorld] = self.multiworld.get_game_worlds("TUNIC")
            # figure out our groups and the items in them
            for tunic in tunic_worlds:
                for group in self.multiworld.get_player_groups(tunic.player):
                    item_link_locations.setdefault(group, {})
            # If there are any TUNIC players in a group, find the locations of all the items belonging to one of the
            # groups.
            if item_link_locations:
                for location in self.multiworld.get_locations():
                    if location.item.player in item_link_locations.keys():
                        locations = item_link_locations[location.item.player].setdefault(location.item.name, [])
                        locations.append((location.player, location.name))
```

Reply:
I did it differently than this -- just checking if groups instead of all of that